blockchain/fullblocktests/generate.go

```diff
@@ -1645,23 +1645,24 @@ func (g *testGenerator) nextBlock(blockName string, spend *spendableOut, ticketS
 	// Create the unsolved block.
 	block := wire.MsgBlock{
 		Header: wire.BlockHeader{
-			Version:     1,
-			PrevBlock:   g.tip.BlockSha(),
-			MerkleRoot:  calcMerkleRoot(regularTxns),
-			StakeRoot:   calcMerkleRoot(stakeTxns),
-			VoteBits:    1,
-			FinalState:  finalState,
-			Voters:      numVotes,
-			FreshStake:  numTicketPurchases,
-			Revocations: numTicketRevocations,
-			PoolSize:    uint32(len(liveTickets)),
-			Bits:        g.calcNextRequiredDifficulty(),
-			SBits:       int64(ticketPrice),
-			Height:      nextHeight,
-			Size:        0, // Filled in below.
-			Timestamp:   ts,
-			Nonce:       0, // To be solved.
-			ExtraData:   [36]byte{},
+			Version:      1,
+			PrevBlock:    g.tip.BlockSha(),
+			MerkleRoot:   calcMerkleRoot(regularTxns),
+			StakeRoot:    calcMerkleRoot(stakeTxns),
+			VoteBits:     1,
+			FinalState:   finalState,
+			Voters:       numVotes,
+			FreshStake:   numTicketPurchases,
+			Revocations:  numTicketRevocations,
+			PoolSize:     uint32(len(liveTickets)),
+			Bits:         g.calcNextRequiredDifficulty(),
+			SBits:        int64(ticketPrice),
+			Height:       nextHeight,
+			Size:         0, // Filled in below.
+			Timestamp:    ts,
+			Nonce:        0, // To be solved.
+			ExtraData:    [32]byte{},
+			StakeVersion: uint32(0),
```

Comment:
No need to cast this.  I realize that some of the other test code below also _incorrectly_ casts things that don't need to be cast, but those tests are self-consistent, so they're acceptable.

However, nothing in this case is casting things that don't need to be (see the `VoteBits`, `Size`, `Nonce`, etc fields), so this should not either.  The entire Decred codebase needs `unconvert` run on it and have all of those unnecessary casts removed.